### PR TITLE
add linux arm build

### PR DESCRIPTION
**Pull request by ReenigneArcher**
## Description

- add linux arm build
  - cross compiled using standard ubuntu packages

## Type of Change
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update (changes to documentation)
- [x] Repository update (changes to repository files)

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated the in code docstring/documentation-blocks for new or existing methods/components

### CODE REVIEW

File: .github/workflows/build-ffmpeg.yml [hunk 1]
```diff
@@ -32,6 +32,20 @@ jobs:
...
+            cmake_generator: Unix Makefiles
+            host: aarch64-unknown-linux-gnu
+            ffmpeg_extras: |
+              --enable-amf \
+              --enable-encoder=h264_amf,hevc_amf \
+              --enable-encoder=h264_nvenc,hevc_nvenc \
+              --enable-encoder=h264_vaapi,hevc_vaapi \
+              --enable-encoder=libx264,libx265 \
+              --enable-nvenc \
+              --enable-vaapi
```
Comment (brad-richardson) [inline .github/workflows/build-ffmpeg.yml:48]: What’s our typical/target host system for this? Is it like a pi or a cloud instance with a GPU? Wondering if we need additional codecs

Comment (ReenigneArcher) [reply .github/workflows/build-ffmpeg.yml:48]: Additional codecs are probably needed. Want to get it to build first, before messing around with other changes though.

**Comment by ReenigneArcher** [comment]
If anyone can help with this it would be appreciated. Currently the linux aarch64 build is failing on the ffmpeg step.

```txt
ERROR: libx264 not found
```

File: .github/workflows/build-ffmpeg.yml (hunk 1 above)
Comment (ReenigneArcher) [reply .github/workflows/build-ffmpeg.yml]: rpi: `h264_v4l2m2m`
jetson_nano: `h264_nvmpi`

File: .github/workflows/build-ffmpeg.yml [hunk 2]
```diff
@@ -164,62 +242,133 @@ jobs:
...
         env:
           root_path: ${{ steps.root.outputs.root_path }}
+          CCPREFIX: ${{ steps.cross.outputs.CCPREFIX }}
         working-directory: ffmpeg_sources/x265_git
         run: |
+          echo "::group::configure extra flags for cross compilation"
+          extra_configure="-DCMAKE_TOOLCHAIN_FILE=build/${{ matrix.arch }}-${{ matrix.os_type }}/crosscompile.cmake"
+          cat ./build/${{ matrix.arch }}-${{ matrix.os_type }}/crosscompile.cmake || true
+          if [[ ${{ matrix.arch }} != "aarch64" ]]; then
+            extra_configure="-DENABLE_HDR10_PLUS=1"
```
Comment (ReenigneArcher) [inline .github/workflows/build-ffmpeg.yml:293]: Todo: will this build aarch64?

File: .github/workflows/build-ffmpeg.yml [hunk 3]
```diff
@@ -164,62 +242,133 @@ jobs:
...
 
       - name: libx265
         env:
           root_path: ${{ steps.root.outputs.root_path }}
+          CCPREFIX: ${{ steps.cross.outputs.CCPREFIX }}
         working-directory: ffmpeg_sources/x265_git
         run: |
+          echo "::group::configure extra flags for cross compilation"
+          extra_configure="-DCMAKE_TOOLCHAIN_FILE=build/${{ matrix.arch }}-${{ matrix.os_type }}/crosscompile.cmake"
+          cat ./build/${{ matrix.arch }}-${{ matrix.os_type }}/crosscompile.cmake || true
```
Comment (ReenigneArcher) [inline .github/workflows/build-ffmpeg.yml:291]: this line can be removed, was only added for testing

File: .github/workflows/build-ffmpeg.yml [hunk 4]
```diff
@@ -164,62 +242,133 @@ jobs:
...
+          echo "$extra_configure"
+          echo "::endgroup::"
+
+          echo "::group::configure"
           PATH="$root_path/bin:$PATH" cmake -G "${{ matrix.cmake_generator }}" \
+            $extra_configure \
             -DCMAKE_INSTALL_PREFIX="$root_path/ffmpeg_build" \
             -DENABLE_CLI=off \
-            -DENABLE_HDR10_PLUS=1 \
+            -DENABLE_SHARED=off \
```
Comment (ReenigneArcher) [inline .github/workflows/build-ffmpeg.yml:303]: test without this flag

File: .github/workflows/build-ffmpeg.yml [hunk 5]
```diff
@@ -232,15 +381,40 @@ jobs:
...
+            || config_error=true
+          set -e  # reset fail on error
+
+          # temp disabled
+          # --ar="${CCPREFIX}ar" \
+          # --cc="${CCPREFIX}gcc" \
+          # --cxx="${CCPREFIX}g++" \
+          # --nm="${CCPREFIX}nm" \
+          # --ranlib="${CCPREFIX}ranlib" \
+          # --ld="${CCPREFIX}ld" \
```
Comment (ReenigneArcher) [inline .github/workflows/build-ffmpeg.yml:395]: remove these, not needed when using cross prefix

File: .github/workflows/build-ffmpeg.yml [hunk 6]
```diff
@@ -232,15 +381,40 @@ jobs:
...
+          echo "::endgroup::"
+
+          if [ $config_error ]
+          then
+            echo "::group::config error log"
+            cat ./ffbuild/config.log
```
Comment (ReenigneArcher) [inline .github/workflows/build-ffmpeg.yml:401]: show the log even when no error

File: .github/workflows/build-ffmpeg.yml (hunk 2 above)
Comment (ReenigneArcher) [reply .github/workflows/build-ffmpeg.yml:293]: `aarch64` is not currently supported: https://bitbucket.org/multicoreware/x265_git/src/cc25ead66daa466ab27cc14897a48a1b61bcf64e/source/dynamicHDR10/CMakeLists.txt#lines-45:46

File: .github/workflows/build-ffmpeg.yml (hunk 1 above)
Comment (ReenigneArcher) [reply .github/workflows/build-ffmpeg.yml:55]: These can be added in a separate PR.

File: .github/workflows/build-ffmpeg.yml [hunk 7]
```diff
@@ -137,6 +208,7 @@ jobs:
             diffutils
             git
             make
+            mercurial
```
Comment (brad-richardson) [inline .github/workflows/build-ffmpeg.yml:211]: What is using mercurial?

**Review by brad-richardson — APPROVED** [review]
🎉🎉🎉

File: .github/workflows/build-ffmpeg.yml (hunk 7 above)
Comment (ReenigneArcher) [reply .github/workflows/build-ffmpeg.yml:211]: I believe it's x265. Not sure what it does but it's optional.

**Pull request merged**